File: src/cz/vutbr/fit/interlockSim/context/EditingContextFactory.java
/* Brno University of Technology
 * Faculty of Information Technology
 * 
 * BSc Thesis	2006/2007
 * 
 * Railway Interlocking Simulator
 * 
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.context;


/**
 * Factory for editing context
 */
public interface EditingContextFactory extends ContextFactory {
    /**
     * create new context
     * @return empty context
     */
    public EditingContext createEmptyContext();
    
    /**
     * @param context
     * @param clazz
     * @param arguments
     * @return create new object in context
     * @throws Exception 
     */
    public Object createNew(EditingContext context, Class<?> clazz, Object... arguments) throws Exception;    
}

File: src/main/kotlin/cz/vutbr/fit/interlockSim/context/EditingContextFactory.kt
/* Brno University of Technology
 * Faculty of Information Technology
 *
 * BSc Thesis  2006/2007
 *
 * Railway Interlocking Simulator
 *
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.context

/**
 * Factory for editing context
 */
interface EditingContextFactory : ContextFactory {
	/**
	 * create new context
	 * @return empty context
	 */
	fun createEmptyContext(): EditingContext

	/**
	 * @param context
	 * @param clazz
	 * @param arguments
	 * @return create new object in context
	 * @throws Exception
	 */
	@Throws(Exception::class)
	fun createNew(
		context: EditingContext,
		clazz: Class<*>,
		vararg arguments: Any
	): Any
}
